### PR TITLE
Openshift: Enable Namespace sync for routes

### PR DESCRIPTION
**Pull request by akshayhavile**
Tested with insecure routes on openshift seutp

### CODE REVIEW

**Comment by akshayhavile** [comment]
build ako

**Review by monotosh-avi — COMMENTED** [review]
Looks good to me. Can we add some UTs too ?

**Comment by akshayhavile** [comment]
@monotosh-avi Sure. Working on it. It will be part of separate PR.

File: internal/k8s/controller.go [hunk 1]
```diff
@@ -294,13 +326,18 @@ func AddRouteEventHandler(numWorkers uint32, c *AviController) cache.ResourceEve
 			newRoute := cur.(*routev1.Route)
 			if isRouteUpdated(oldRoute, newRoute) {
 				namespace, _, _ := cache.SplitMetaNamespaceKey(utils.ObjKey(newRoute))
-				key := utils.OshiftRoute + "/" + utils.ObjKey(newRoute)
-				bkt := utils.Bkt(namespace, numWorkers)
-				if !lib.HasValidBackends(newRoute.Spec, newRoute.Name, namespace, key) {
-					status.UpdateRouteStatusWithErrMsg(newRoute.Name, namespace, lib.DuplicateBackends)
+				nsFilterObj := utils.GetGlobalNSFilter()
+				if utils.CheckIfNamespaceAccepted(namespace, nsFilterObj, nil, true) {
```
Comment (parimanur) [inline internal/k8s/controller.go:330]: Nit: Can you move the return logic and remove the else?


```
if !utils.CheckIfNamespaceAccepted(namespace, nsFilterObj, nil, true) {
// log and return
}
// enqueue to work queue. 
```
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

**Comment by akshayhavile** [comment]
Review comments addressed

**Review by monotosh-avi — APPROVED** [review]

**Review by parimanur — APPROVED** [review]

**Pull request merged**